### PR TITLE
Drop Greenlet to GCCcore

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
n1441 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/c9b9147a46685b069bb8afd7af39bdb4 for a full test report.

File: easybuild/easyconfigs/g/Greenlet/Greenlet-2.0.2-GCCcore-11.3.0.eb [hunk 1]
```diff
@@ -12,7 +12,7 @@ A "greenlet", on the other hand, is a still more primitive notion of micro-threa
 scheduling; coroutines, in other words. This is useful when you want to control exactly when your code runs.
 """
 
-toolchain = {'name': 'foss', 'version': '2022a'}
+toolchain = {'name': 'GCCcore', 'version': '11.3.0'}
 
 dependencies = [('Python', '3.10.4')]
```
Comment (Micket) [inline easybuild/easyconfigs/g/Greenlet/Greenlet-2.0.2-GCCcore-11.3.0.eb:17]:
```suggestion
builddependencies = [('binutils', '2.39')]
dependencies = [('Python', '3.10.4')]
```

File: easybuild/easyconfigs/g/Greenlet/Greenlet-2.0.2-GCCcore-11.3.0.eb [hunk 2]
```diff
@@ -12,8 +12,9 @@ A "greenlet", on the other hand, is a still more primitive notion of micro-threa
 scheduling; coroutines, in other words. This is useful when you want to control exactly when your code runs.
 """
 
-toolchain = {'name': 'foss', 'version': '2022a'}
+toolchain = {'name': 'GCCcore', 'version': '11.3.0'}
 
+builddependencies = [('binutils', '2.39')]
```
Comment (Flamefire) [inline easybuild/easyconfigs/g/Greenlet/Greenlet-2.0.2-GCCcore-11.3.0.eb:17]:
```suggestion
builddependencies = [('binutils', '2.38')]
```

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
n1265 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/7bb782ab9a3c656961bf259e24222978 for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/34bcab4f4a479b8f84295dc80fff3535 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 49 out of 49 (3 easyconfigs in total)
i7006 - Linux Rocky Linux 8.7 (Green Obsidian), x86_64, AMD EPYC 7702 64-Core Processor (zen2), Python 3.8.13
See https://gist.github.com/Flamefire/6455076a0eb18659ba2c9d4be16165fa for a full test report.

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen3

**Pull request merged**